evi-typescript-chat-history-example/src/index.ts

```diff
@@ -0,0 +1,160 @@
+import fs from "fs"; 
+import dotenv from "dotenv";
+import { HumeClient } from "hume";
+import { ReturnChatEvent, EmotionScores } from "hume/api/resources/empathicVoice";
+
+dotenv.config();
+
+/**
+ * The main entry point of the script.
+ * 
+ * Steps:
+ *  1. Fetch all chat events for the specified chat ID.
+ *  2. Generate a transcript from user and assistant messages.
+ *  3. Save the transcript to a local text file.
+ *  4. Calculate and log the top 3 emotions (by average score) from user messages.
+ */
+async function main(): Promise<void> {
+  const CHAT_ID = "<YOUR_CHAT_ID>"; // Replace with your actual Chat ID
+
+  try {
+    const chatEvents = await fetchAllChatEvents(CHAT_ID);
+
+    // Generate a transcript string from the fetched chat events
+    const transcript = generateTranscript(chatEvents);
+
+    // Define the transcript file name
+    const transcriptFileName = `transcript_${CHAT_ID}.txt`;
+
+    // Write the transcript to a text file
+    try {
+      fs.writeFileSync(transcriptFileName, transcript, "utf8");
+      console.log(`Transcript saved to ${transcriptFileName}`);
+    } catch (fileError) {
+      console.error(`Error writing to file ${transcriptFileName}:`, fileError);
+    }
+
+    // Calculate and log the top 3 emotions (on average)
+    const topEmotions = getTopEmotions(chatEvents);
+    console.log("Top 3 Emotions:", topEmotions);
+  } catch (error) {
+    console.error("An error occurred:", error);
+  }
+}
+
+main();
```

Comment:
nit: I more commonly see
```suggestion
main().catch((err) => console.log("An error occurred:", error))
```
instead of an outer-level try/catch